storage/range_command.go

```diff
@@ -681,6 +681,8 @@ func (r *Range) InternalLeaderLease(batch engine.Engine, ms *proto.MVCCStats, ar
 	// extra tick. This allows multiple requests from the same replica to
 	// merge without ticking away from the minimal common start timestamp.
 	if prevLease.RaftNodeID == 0 || isExtension {
+		// TODO(tschottdorf) probably go all the way back to
+		// prevLease.Start() (so it's properly extending previous lease).
```

Comment:
I'm not sure I understand the TODO since this case is explicitly that the previous lease didn't exist.